[TensorPipe] Pass names of endpoints to context/pipe for easier debugging

lw opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #39012 [RPC] Fix flaky test by waiting for async rref calls
* #39011 [TensorPipe] Detect duplicate worker names
* #39010 [TensorPipe] Improve serialization
* #38934 [TensorPipe] Close and join TP context at shutdown
* #38933 [TensorPipe] Implement join correctly
* #38930 [TensorPipe] Always complete futures from thread pool
* #38931 [TensorPipe] Do not mark future messages as complete after they have timed out
* #38929 [TensorPipe] Fix lock inversion upon response read error handling
* #38928 [TensorPipe] Fix timeout computation
* #38927 [TensorPipe] Add cases for TP in RPC test helpers
* **#38926 [TensorPipe] Pass names of endpoints to context/pipe for easier debugging**



TensorPipe supports for the user to provide a meaningful name for each context and to specify what it thinks the name of the endpoint it's connecting to is, so that these names can be logged and matched to the otherwise not-very-informative ID of a pipe (given by the PID and some counters) for easier debugging.

Differential Revision: [D21479799](https://our.internmc.facebook.com/intern/diff/D21479799/)

Differential Revision: [D21479799](https://our.internmc.facebook.com/intern/diff/D21479799)

lw committed:
[TensorPipe] Pass names of endpoints to context/pipe for easier debugging

TensorPipe supports for the user to provide a meaningful name for each context and to specify what it thinks the name of the endpoint it's connecting to is, so that these names can be logged and matched to the otherwise not-very-informative ID of a pipe (given by the PID and some counters) for easier debugging.

Differential Revision: [D21479799](https://our.internmc.facebook.com/intern/diff/D21479799/)

[ghstack-poisoned]

lw committed:
Update on "[TensorPipe] Pass names of endpoints to context/pipe for easier debugging"




TensorPipe supports for the user to provide a meaningful name for each context and to specify what it thinks the name of the endpoint it's connecting to is, so that these names can be logged and matched to the otherwise not-very-informative ID of a pipe (given by the PID and some counters) for easier debugging.

Differential Revision: [D21479799](https://our.internmc.facebook.com/intern/diff/D21479799/)

Differential Revision: [D21479799](https://our.internmc.facebook.com/intern/diff/D21479799)

[ghstack-poisoned]

mrshenli reviewed (approved):
The change LGTM. Could you please rebase to https://github.com/pytorch/pytorch/commit/916084d933792f7ee619aee7155fedf68d1a8cd1 to get the test signals clear?

mrshenli reviewed (changes requested):
Adding "request changes" only because wanted to see clear test signals.

lw committed:
Update on "[TensorPipe] Pass names of endpoints to context/pipe for easier debugging"




TensorPipe supports for the user to provide a meaningful name for each context and to specify what it thinks the name of the endpoint it's connecting to is, so that these names can be logged and matched to the otherwise not-very-informative ID of a pipe (given by the PID and some counters) for easier debugging.

Differential Revision: [D21479799](https://our.internmc.facebook.com/intern/diff/D21479799/)

Differential Revision: [D21479799](https://our.internmc.facebook.com/intern/diff/D21479799)

[ghstack-poisoned]

mrshenli reviewed (approved):
Test failures are irrelevant.
